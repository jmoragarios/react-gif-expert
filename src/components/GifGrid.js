import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect( () => {
        getGifs();
    }, []);

    const getGifs = async () => {
        
        const url = 'http://api.giphy.com/v1/gifs/search?q=One%20Piece&limit=10&api_key=F5av68ecrjZl4tx7XtSN679Z8HVx91Bl';
        const response = await fetch( url );
        const { data } = await response.json();
        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_large.url
            }
        })
        
        console.log(gifs);
        setImages(gifs);
    }

    return (
        <>
            <h3>{ category }</h3>

            {
                images.map( img => (
                    <GifGridItem
                        key={img.id}
                        {...img}    
                    />
                ))
            }
            
        </>
    )
}
