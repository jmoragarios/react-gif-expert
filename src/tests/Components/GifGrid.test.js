import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Tests in <GifGrid/>', () => {
    
    const category = 'One Pounch';

    test('should show the component successfully', () => {
        
        // simulate hook run
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category }/>);

        expect( wrapper ).toMatchSnapshot();
    });

    test('should show items when images are loaded from useFetchGifs', () => {
        
        // dummy data
        const gifs = [
            {
            id: 'ABC',
            url: 'https://localhost/any/thing.jpg',
            title: 'Any thing'
        },
        {
            id: '123',
            url: 'https://localhost/any/thing.jpg',
            title: 'Any thing'
        }
    ]

        // simulate hook run
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ category }/>);

        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    })
    
    
})
