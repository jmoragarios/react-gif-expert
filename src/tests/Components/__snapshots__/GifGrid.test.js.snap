// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGrid/> should show the component successfully 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    One Pounch
  </h3>
  <p
    className="card animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
